### ISSUE TITLE
Bounty Redeemed event, ISF errors

### ISSUE THREAD

**Issue by Darkcyde13** [EDCD/EDDI#298]
## EDDI version in which issue found
[2.4.6-b1]

## Steps to reproduce
Hand in some bounties at Interstellar Factors

## Expected
Bounty Redeemed event speaks the correct information.

## Observed
No factions are spoken and broker percentage is the actual percent, not the amount of credits.

## Investigation

This is what I had spoken when handing in some bounty vouchers at Tiliala's Lament in Varpet:-
> Bounty vouchers for Over 268 thousand credits redeemed by Bedaho Major papa lima charlie, Varpet Industries and Federation.
> Bounty vouchers for Nearly 602 thousand credits redeemed by and.  The Broker has taken 25 credits.

The first was 'Hand all in' at the normal 'Authority Contact'.  The second was 'Hand all in' at ISF.  The first is fine, but the second is missing the factions, and the percentage is not the credit amount.

Here are the journal entries.

```
{ "timestamp":"2017-12-10T02:11:55Z", "event":"RedeemVoucher", "Type":"bounty", "Amount":268541, "Factions":[ { "Faction":"Bedaho Major PLC", "Amount":49262 }, { "Faction":"Varpet Industries", "Amount":79000 }, { "Faction":"Federation", "Amount":140279 } ] }
{ "timestamp":"2017-12-10T02:12:12Z", "event":"RedeemVoucher", "Type":"bounty", "Amount":601644, "Factions":[ { "Faction":"", "Amount":46087 }, { "Faction":"", "Amount":555557 } ], "BrokerPercentage":25.000000 }
```

As you can see, when handing in the vouchers at ISF, the journal event is empty for the factions (so not an EDDI bug), and the broker percentage is just that, the percent value.

Obviously, there is nothing to be done to fix the missing factions (other than tell FDev), but the broker part of the script can be fixed by making changes the following line:-
`The Broker has taken {round((event.amount / (100 - event.brokerpercentage)) * event.brokerpercentage, 0)} credits.
`
This allows for variations in the broker percentage, rather than fixed at 25%.

**Comment by Darkcyde13** [comment]
Having thought about it, you could just add a new check within the Occasionally() to check if the broker percentage has been set.  Here's the whole script:-

```
{_ Context }
{SetState('eddi_context_last_subject', 'bounty')}
{SetState('eddi_context_last_action', 'redeem')}
{SetState('eddi_context_bounty_amount', event.amount)}

Bounty vouchers for {Humanise(event.amount)} credits redeemed
{Occasionally(2, "
   {if !event.brokerpercentage:
      {if len(event.rewards) = 1:
         by {P(event.rewards[0].faction)}
      |else:
         by
        {set cur to 0}
        {while cur < len(event.rewards):
           {if cur = 0:
           |elif cur < len(event.rewards) - 1:
              , 
           |else:
              and
           }
           {P(event.rewards[cur].faction)}
           {set cur to cur + 1}
       }
     }
   }
")}.

{if event.brokerpercentage:
   {Pause(1000)}
   The Broker has taken {round((event.amount / (100 - event.brokerpercentage)) * event.brokerpercentage, 0)} credits.
}

```

**Comment by Tkael** [comment]
Proposed:
```
{_ Context }
{SetState('eddi_context_last_subject', 'bond')}
{SetState('eddi_context_last_action', 'redeem')}
{SetState('eddi_context_bond_amount', event.amount)}

Combat bonds for {Humanise(event.amount)} credits 
{Occasionally(2, OneOf("have been", "have now been"))}
{OneOf("redeemed", "claimed")}

{Occasionally(2, "
  {if len(event.rewards[0].faction) = 1:
    from {P(event.rewards[0].faction)}
  |else:
    . 
    {set cur to 0}
    {while cur < len(event.rewards[cur].faction):
        {if cur = 0:
        |elif cur < len(event.rewards) - 1:
            , 
        |else:
            and
        }
        {Humanise(event.rewards[cur].amount)} credits from
        {P(event.rewards[cur].faction)} 
        {set cur to cur + 1}
    }
  }
")}.

{if event.brokerpercentage && event.brokerpercentage > 0:
   {OneOf("Your", "The")}
   {OneOf("broker's", "contact's")}
   {OneOf("payment", "fee", "share")}
   {OneOf("was", "came to")}
   {round((event.amount / (100 - event.brokerpercentage)) * event.brokerpercentage, 0)} credits.
}
```

**Comment by Tkael** [comment]
Closed pending approval of #336.
